src/main/java/com/google/devtools/build/lib/syntax/MethodDescriptor.java

```diff
@@ -163,13 +141,37 @@ public Object call(Object obj, Object[] args, Location loc, Environment env)
       if (e.getCause() instanceof FuncallExpression.FuncallException) {
         throw new EvalException(loc, e.getCause().getMessage());
       } else if (e.getCause() != null) {
+        Throwables.throwIfUnchecked(e.getCause());
         Throwables.throwIfInstanceOf(e.getCause(), InterruptedException.class);
```

Comment:
It looks like we can use `propagateIfPossible` to combine the two throw methods.

Reply:
sgtm

Reply:
BTW, do you think this should also be rethrowing `EvalException`? It doesn't seem very useful to rewrap it.

Reply:
I think the important part is the location - the original EvalException might not have it

Reply:
Should it use `EvalUtils.ensureLocation()`?

Reply:
Yes, good idea!